### DISCUSSION TITLE
React Native Web Component Testing

### DISCUSSION THREAD

**Discussion by mbrimmer83** [cypress-io/cypress#24389] [Questions and Help]
We are already using cypress to test react-native-web sites and it has served us well. Is there a way to compile react native components using react-native-web for component testing? Could the component be compiled before it is mounted?

Thanks
Matthew

**Comment by lmiller1990** [comment]
I am not familiar with React Native Web - this should be possible, though, how do you normally compile RNW? Just with a webpack loader or something? Or is it more involved?

**Reply by mbrimmer83** [reply]
Yes, bundles get compiled with Webpack and I think babel just swaps out the `react-native` component imports to the corresponding `react-native-web` component.

**Reply by lmiller1990** [reply]
In this case nothing should stop you from wiring it up.

Here's a simple example showing how to make a mount function for Web Components: https://github.com/lmiller1990/cypress-web-components-example

We can no doubt do something similar for React Native Web.
